crates/astria-conductor/src/executor.rs

```diff
@@ -273,47 +301,56 @@ impl<C: ExecutionClientV1Alpha1> Executor<C> {
                 // execute_block will check if our namespace has txs; if so, it'll return the
                 // resulting execution block hash, otherwise None
                 let Some(execution_block_hash) = self
-                    .execute_block(block)
+                    .execute_block(block.clone())
                     .await
                     .wrap_err("failed to execute block")?
                 else {
                     // no txs for our namespace, nothing to do
-                    debug!("execute_block returned None; skipping finalize_block");
+                    debug!("execute_block returned None; skipping call_update_commitment_state");
                     return Ok(());
                 };
 
-                // finalize the block after it's been executed
-                self.finalize_block(execution_block_hash, sequencer_block_hash)
-                    .await?;
+                self.handle_executed_da_block(block, execution_block_hash)
```

Comment:
I don't think this case is handling updating the soft commitment, because in this case we executed & have firm commitment we need to update both state blocks.